File: Baekjun/9.2/17947. 상남자 곽철용/solution.swift
let NMK = readLine()!.split(separator: " ").compactMap {Int($0)}
let N = NMK[0], M = NMK[1], K = NMK[2]
var card = Array(repeating: 1, count: 4*N + 1)

for _ in 0..<M {
    let ab = readLine()!.split(separator: " ").compactMap {Int($0)}
    card[ab[0]] -= 1
    card[ab[1]] -= 1
}

let cy = readLine()!.split(separator: " ").compactMap{Int($0)}

card[cy[0]] -= 1
card[cy[1]] -= 1

let gap = abs(cy[0]%K - cy[1]%K)

card = (1..<4*N+1).filter {card[$0] == 1}

for i in 0..<card.count {
    card[i] %= K
}

card.sort()

var ans = 0
var l = 0, r = 0

while (r < card.count && ans < M - 1) {

    if card[r] - card[l] <= gap {
        r += 1
    } else {
        ans += 1
        r += 1
        l += 1
    }
}

print(ans)

File: Baekjun/9.2/17947. 상남자 곽철용/solution.py
import sys

input = sys.stdin.readline

N, M, K = map(int, input().split())

card = [1] * (4 * N + 1)

for _ in range(M):
    a, b = map(int, input().split())
    card[a] = 0
    card[b] = 0


chulyong = list(map(int, input().split()))
card[chulyong[0]] = 0
card[chulyong[1]] = 0

chulyong_score = abs(chulyong[0] % K - chulyong[1] % K)

card = [i for i in range(1, 4 * N + 1) if card[i]]

for i in range(len(card)):
    card[i] %= K

card.sort()

left = right = answer = 0

while right < len(card) and answer < M - 1:
    if card[right] - card[left] > chulyong_score:
        answer += 1
        right += 1
        left += 1
    else:
        right += 1

print(answer)
